### DISCUSSION TITLE
Allow empty default group

### DISCUSSION THREAD

**Discussion by ECode16** [codeigniter4/shield#1062] [Ideas]
If a defaultGroup in AuthGroups is empty/null just skip it instead of throwing.

**Comment by datamweb** [comment]
Hi, please explain what you need for it?

**Reply by ECode16** [reply]
My users are divided into several "basic" groups whose permissions don't intersect. (Lets say its by a continent and they can only affect things on their own continent.)

**Comment by kenjis** [comment]
The default group is only the first group set by `RegisterController`.
If you want to set a different group, just override the `addToDefaultGroup()` method.
